### PR TITLE
Two Windows-related doc/CMakeLists.txt fixes

### PR DESCRIPTION
**Pull request by carlescufi**
See commit messages

### CODE REVIEW

**Review by mbolivar — APPROVED** [review]
Commit message for moving python finding up earlier could use a note about how version.cmake is being moved in case it ever depends on python in the future, but that's minor.

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@ if(${WEST} STREQUAL WEST-NOTFOUND)
   unset(WEST)
 endif()
 
-set(ZEPHYR_BASE $ENV{ZEPHYR_BASE})
+file(TO_CMAKE_PATH $ENV{ZEPHYR_BASE} ZEPHYR_BASE)
```
Comment (SebastianBoe) [inline doc/CMakeLists.txt]: We should use the same mechanism as in a normal build. So either apply this change also the normal build, or use the below mechanism in the doc build:

string(REPLACE "\\" "/" ZEPHYR_BASE ${ZEPHYR_BASE})

Comment (carlescufi) [reply doc/CMakeLists.txt]: I will apply the same mechanism to the normal build then

**Comment by carlescufi** [comment]
> Commit message for moving python finding up earlier could use a note about how version.cmake is being moved in case it ever depends on python in the future, but that's minor.

Added, thanks

File: doc/CMakeLists.txt (hunk 1 above)
Comment (carlescufi) [reply doc/CMakeLists.txt]: Fixed

**Review by SebastianBoe — APPROVED** [review]

**Pull request merged**